SU2_CFD/include/variables/CAdjEulerVariable.hpp

```diff
@@ -167,4 +167,10 @@ class CAdjEulerVariable : public CVariable {
    */
   inline su2double **GetGradient_Reconstruction(unsigned long iPoint) final { return Gradient_Reconstruction[iPoint]; }
 
+  /*!
+   * \brief Get the reconstruction gradient for variables at all points.
+   * \return Reference to reconstruction gradient.
+   */
+  inline VectorOfMatrix& GetGradient_Reconstruction(void) final { return Gradient_Reconstruction; }
+
```

Comment:
To allow the new routines to work on primitives / solution CVariable needs to give access to the entire container... Not because the routine depends on a specific type of container, but to bypass the names of the AddGradient_XX SubtractGradient_XX methods.
To eventually vectorize the gradient and limiter computation only the container will need to implement "vector-accessors".